File: app/src/main/java/com/smallcake/temp/weight/NotTouchScrollView.java
package com.smallcake.temp.weight;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.RelativeLayout;

import androidx.core.widget.NestedScrollView;

/**
 * Author: Relin
 * Describe:ScrollView滑动冲突
 * Date:2020/5/7 9:43
 */
public class NotTouchScrollView extends RelativeLayout {

    private NestedScrollView scrollView;

    public NotTouchScrollView(Context context) {
        super(context);
    }

    public NotTouchScrollView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public NotTouchScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setScrollView(NestedScrollView scrollView) {
        this.scrollView = scrollView;
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (scrollView==null)return false;
        if (ev.getAction() == MotionEvent.ACTION_UP) {
            scrollView.requestDisallowInterceptTouchEvent(false);
        } else {
            scrollView.requestDisallowInterceptTouchEvent(true);
        }
        return false;
    }
    @Override
    public boolean onTouchEvent(MotionEvent event) {
        return true;
    }


}

File: app/src/main/java/com/smallcake/temp/weight/NotTouchScrollView.kt
package com.smallcake.temp.weight

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.RelativeLayout
import androidx.core.widget.NestedScrollView

/**
 * Describe:ScrollView滑动冲突
 */
class NotTouchScrollView : RelativeLayout {
    private var scrollView: NestedScrollView? = null

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
    }

    fun setScrollView(scrollView: NestedScrollView?) {
        this.scrollView = scrollView
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        if (scrollView == null) return false
        if (ev.action == MotionEvent.ACTION_UP) {
            scrollView!!.requestDisallowInterceptTouchEvent(false)
        } else {
            scrollView!!.requestDisallowInterceptTouchEvent(true)
        }
        return false
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        return true
    }
}